src/background.js

```diff
@@ -34,29 +36,371 @@
+var DomainExceptions = require("domainExceptions").DomainExceptions;
+var HeuristicBlocking = require("heuristicblocking");
+var SocialWidgetLoader = require("socialwidgetloader");
+var pbStorage = require("storage");
+var webrequest = require("webrequest");
+var SocialWidgetList = SocialWidgetLoader.loadSocialWidgetsFromFile("src/socialwidgets.json");
 
-// Add a permanent store for seen third parties 
-var seenCache = localStorage.getItem("seenThirdParties");
-var seenThirdParties = JSON.parse(seenCache);
-if (!seenThirdParties){
-  localStorage.setItem("seenThirdParties", JSON.stringify({}));
-  seenThirdParties = {};
-}
+var pb = {
+  // imports
+  heuristicBlocking: HeuristicBlocking,
+  utils: Utils,
+  storage: pbStorage,
+  webrequest: webrequest,
+  
+  // Tracking status constants
+  NO_TRACKING: "noaction",
+  ALLOW: "allow",
+  BLOCK: "block",
+  COOKIEBLOCK: "cookieblock",
+  DNT: "dnt",
+  USER_ALLOW: "user_allow",
+  USER_BLOCK: "user_block",
+  USER_COOKIE_BLOCK: "user_cookieblock",
+
+  // URLS
+  DNT_POLICIES_URL: "https://www.eff.org/files/dnt-policies.json",
+  COOKIE_BLOCK_LIST_URL: "https://www.eff.org/files/cookieblocklist_new.txt",
+
+  // The number of 1st parties a 3rd party can be seen on
+  TRACKING_THRESHOLD: 3,
+  MAX_COOKIE_ENTROPY: 12,
+  
+  // Display debug messages
+  DEBUG: true,
+  INITIALIZED: false,
+  
+  /**
+  * Per-tab data that gets cleaned up on tab closing
+    looks like:
+      tabData = {
+        <tab_id>: {
+          fpData: {
+            <script_origin>: {
+              canvas: {
+                fingerprinting: boolean,
+                write: boolean
+              }
+            },
+            ...
+          },
+          frames: {
+            <frame_id>: {
+              url: string,
+              parent: int
+            },
+            ...
+          },
+          trackers: {
+            domain.tld: bool
+            ...
+          }
+        },
+        ...
+      } 
+  */
+  tabData: {},
 
-setInterval(function(){
-  if(seenCache != localStorage.getItem("seenThirdParties")) {
-    seenCache = localStorage.getItem("seenThirdParties");
-    seenThirdParties = JSON.parse(seenCache);
-  }
-}, 1000);
 
+  // Methods
+  /**
+   * initialize privacy badger
+   */
+  init: function(){
+    if(pb.INITIALIZED) { return; }
+
+    pb.storage.initialize();
+    pb.updateTabList();
+    pb.initializeCookieBlockList();
+    pb.initializeDNT();
+    pb.initializeDefaultSettings();
+
+    // Show icon as page action for all tabs that already exist
+    chrome.windows.getAll({populate: true}, function(windows) {
+      for (var i = 0; i < windows.length; i++) {
+        for (var j = 0; j < windows[i].tabs.length; j++) {
+          refreshIconAndContextMenu(windows[i].tabs[j]);
+        }
+      }
+    });
+    pb.showFirstRunPage();
+    
+    pb.INITIALIZED = true;
+    console.log('privacy badger is ready to rock');
+    console.log('set pb.DEBUG=1 to view console messages');
```

Comment:
Indeed I have had it set to true for development purposes but we should set it back to false before we release.